cycles-ledger/src/storage.rs

```diff
@@ -1261,25 +1288,7 @@ fn process_block(
         .map(|fee| effective_fee.or(fee))
         .map_err(|expected_fee| PTErr::BadFee { expected_fee })?;
 
-    // sanity check that the transaction can be hashed
-    let tx_hash = match transaction.hash() {
-        Ok(tx_hash) => tx_hash,
-        Err(err) => {
-            let err = err.context(format!("Unable to process transaction {:?}", transaction));
-            log!(P0, "{:#}", err);
-            return Err(PTErr::from(err));
-        }
-    };
-
-    // check whether transaction is a duplicate
-    if let Some((block_index, maybe_canister)) =
-        read_state(|state| state.transaction_hashes.get(&tx_hash))
-    {
-        return Err(PTErr::Duplicate {
-            duplicate_of: block_index,
-            canister_id: maybe_canister.map(Into::into),
-        });
-    }
+    check_duplicate(&transaction)?;
```

Comment:
```suggestion
```
Did we decide if we want to remove the duplicate check in `process_block`? IMO we don't need the check just in case someone adds a new flow because they will most likely draw inspiration from the existing ones

Reply:
removed